src/System.Private.Xml/src/System/Xml/Serialization/Compilation.cs

```diff
@@ -167,29 +167,49 @@ internal static Assembly LoadGeneratedAssembly(Type type, string defaultNamespac
                 name.Name = serializerName;
                 name.CodeBase = null;
                 name.CultureInfo = CultureInfo.InvariantCulture;
-                string serializerPath = Path.Combine(Path.GetDirectoryName(type.Assembly.Location), serializerName + ".dll");
-                if (!File.Exists(serializerPath))
+
+                string serializerPath = null;
+
+                try
+                {
+                    serializerPath = Path.Combine(Path.GetDirectoryName(type.Assembly.Location), serializerName + ".dll");
+                }
+                catch
```

Comment:
Any particular reason we wrap each `Path.Combine(...)` in separate `try...catch`s? Can we put all in the same `try...catch` as the `serializer = Assembly.LoadFile(serializerPath)`?

Reply:
You can still call the second Path.Combine if it fail in the first Path.Combine.

Reply:
With null check you added, I think it's safe to put them all in one `try...catch` now.